Ft2023 341 : Created a simple and config form

sandipta18 opened the pull request.

vivekpan27 commented:
Please update this branch with latest code from main branch.
I still see 600 file changes.

sandipta18 commented:
1. I have reduced the files from 600 to 400 is it ok now?
2. While installing the Drupal website there are two options one is installing the normal website and another is installing a test website I have installed that, and hence after exporting configurations, the number of files changed is generally higher for me

sandipta18 commented:
Fixed @vivekpan27

vivekpan27 commented on .editorconfig:
![Screenshot from 2023-07-11 16-39-36](https://github.com/sandipta18/Drupal-Backend/assets/25786248/2fbe1861-410a-4b0c-8aac-41271d73c3e4)

These files are already in the main branch.
how are they showing up in the PR?

sandipta18 replied:
Now is it ok?

vivekpan27 replied:
What do you think?

sandipta18 replied:
These files are not present on my on my current PR and I followed exactly the same thing you mentioned, i.e. updating the local main with the latest merged code and then on switched to my current branch and merging the main here.
Now regarding the previous modules showing up, I did a mistake in previous modules in core_version_requirement and you said to address them in the next PR that's why I changed them here.

vivekpan27 commented:
How were you able to fix it?

sandipta18 commented:
Actually previously i pushed all of my code, that's why the number of files changed was so high. Now currently I only pushed the file/folder that was modified. And hence it was fixed

vivekpan27 commented:
> Actually previously i pushed all of my code, that's why the number of files changed was so high. Now currently I only pushed the file/folder that was modified. And hence it was fixed

Did you do as I mentioned?
That is push the latest main branch code to the feature branch?

sandipta18 commented:
Yes i pulled the main branch code on the feature branch. And additionally i am also maintaining it locally so there's that.

sandipta18 commented:
I have made a few changes to the interface i.e. deleted a few footers and blocks here and there and that's why it was reflected here in the yml file.
Mix is a module and I have turned on a few features of it that's why it was reflected here in the yml file.
Implemented different permissions that's why it was also reflected here in the yml file.
Changed the core_version_requirement form 10 to ^10 that's why it was also reflected in the yml file.
Installed a new module for the configform task that's why the core.extension.yml was updated and hence the changes were reflected.
That's the reason these files are present in the PR.
Is there more? Am i misising something ? Please let me know. @vivekpan27

vivekpan27 commented on web/modules/custom/configform/configform.routing.yml:
```diff
@@ -0,0 +1,15 @@
+Form.simpleForm:
```
we don't start with capital letter

vivekpan27 commented on web/modules/custom/configform/configform.routing.yml:
```diff
@@ -0,0 +1,15 @@
+Form.simpleForm:
+  path: '/form'
```
this path is not good.
make it more meaninful.

vivekpan27 commented on web/modules/custom/configform/configform.routing.yml:
```diff
@@ -0,0 +1,15 @@
+    _permission: 'simple custom form permission'
+
+Form.configForm:
+  path: '/admin/config/form'
```
this path is not good.
make it more meaninful.

vivekpan27 commented on web/modules/custom/configform/src/Form/ConfigForm.php:
```diff
@@ -0,0 +1,259 @@
+      '#type'      => 'markup',
+      '#markup'    => '<div id="error-message"></div>',
+    ];
+    $form['FullName'] = [
```
Why are the keys beginning with capital letter?

vivekpan27 commented on web/modules/custom/configform/src/Form/ConfigForm.php:
```diff
@@ -0,0 +1,259 @@
+    $form['Subscribe'] = [
+      '#type'     => 'radios',
+      '#title'    => $this->t('Subscribe'),
+      '#required' => TRUE,
```
There should be a default value for this

vivekpan27 commented on web/modules/custom/configform/src/Form/ConfigForm.php:
```diff
@@ -0,0 +1,259 @@
+        ],
+      ],
+    ];
+    $form['action']['submit'] = [
```
Since this is the config form, where are we updating the configs?

sandipta18 replied:
Added

vivekpan27 commented on web/modules/custom/configform/src/Form/SimpleForm.php:
```diff
@@ -0,0 +1,151 @@
+   */
+  public function __construct(MessengerInterface $messenger, RequestStack $request_stack) {
+    $this->messenger = $messenger;
+    $this->requestStack = $request_stack;
```
Where is `$this->requestStack` defined?

vivekpan27 commented:
> I have made a few changes to the interface i.e. deleted a few footers and blocks here and there and that's why it was reflected here in the yml file. Mix is a module and I have turned on a few features of it that's why it was reflected here in the yml file. Implemented different permissions that's why it was also reflected here in the yml file. Changed the core_version_requirement form 10 to ^10 that's why it was also reflected in the info.yml files of other modules. Installed a new module for the configform task that's why the core.extension.yml was updated and hence the changes were reflected. That's the reason these files are present in the PR. Is there more? Am i misising something ? Please let me know. @vivekpan27

I was only talking about the files which I shared in the screenshot in my previous comment.
I see you have deleted the files after I made the comment in this commit https://github.com/sandipta18/Drupal-Backend/pull/4/commits/89730856ace4fd4f0b920c6c7c76e714de478fdc.
But that's not proper.
When you merged the main branch, those files should have been removed automatically from your PR, but you removed them manually.
Try to understand the flow here.
Anyways, I have reviewed again and added my comments.

vivekpan27 reviewed (changes requested):
Please check.

sandipta18 commented:
> > I have made a few changes to the interface i.e. deleted a few footers and blocks here and there and that's why it was reflected here in the yml file. Mix is a module and I have turned on a few features of it that's why it was reflected here in the yml file. Implemented different permissions that's why it was also reflected here in the yml file. Changed the core_version_requirement form 10 to ^10 that's why it was also reflected in the info.yml files of other modules. Installed a new module for the configform task that's why the core.extension.yml was updated and hence the changes were reflected. That's the reason these files are present in the PR. Is there more? Am i misising something ? Please let me know. @vivekpan27
> 
> I was only talking about the files which I shared in the screenshot in my previous comment. I see you have deleted the files after I made the comment in this commit [8973085](https://github.com/sandipta18/Drupal-Backend/commit/89730856ace4fd4f0b920c6c7c76e714de478fdc). But that's not proper. When you merged the main branch, those files should have been removed automatically from your PR, but you removed them manually. Try to understand the flow here. Anyways, I have reviewed again and added my comments.

Sir i haven't removed all the files manually. What happened here is previously all the files and folders were pushed on github, now after updating local main with remote main and merging the main into branch, only the updated stuff was pushed, and hence those files were marked deleted(since they were already pushed and afterwards it was removed).

You can check my vscode, i haven't manually removed them, they are still present. And i am also tracking them. 
I am attaching a ss listing out the files and folders i am tracking using command `git ls-tree --full-tree --name-only -r HEAD`

![Screenshot from 2023-07-12 11-25-56](https://github.com/sandipta18/Drupal-Backend/assets/122454028/30731a7b-cb99-4548-8f43-d9f5089dd33b)
![Screenshot from 2023-07-12 11-25-52](https://github.com/sandipta18/Drupal-Backend/assets/122454028/6abb9779-1a90-494f-97ce-c81359cd43ab)
![Screenshot from 2023-07-12 11-24-22](https://github.com/sandipta18/Drupal-Backend/assets/122454028/1f91eb5c-e8a1-49aa-9a66-50b562ffbb65)

sandipta18 commented on config/sync/.htaccess:
Will fix it

vivekpan27 commented:
PR is fine now.
But default configuration is missing for the module.
Please check how to add default configurations for the module.

vivekpan27 commented on web/modules/custom/configform/src/Form/SimpleForm.php:
```diff
@@ -0,0 +1,142 @@
+  public function submitData(array &$form, FormStateInterface $form_state) {
+    $ajax_response = new AjaxResponse();
+    $values = $form_state->getValues();
+    \Drupal::database()->insert('configform_example')->fields([
```
Did you check your code in PHPCS before raising this PR?

sandipta18 replied:
Yes, it suggested using Dependency injection. Now the issue is Ticket FT2023-344 is about updating old code and implementing dependency injection there. So I have kept this pr as it is, and implemented dependency injection on ticket 344,

sandipta18 replied:
Please let me know if I should push the dependency injection code in this branch only or I should keep it for the FT2023-344 branch

sandipta18 commented on web/modules/custom/configform/config/schema/configform.schema.yml:
```diff
@@ -0,0 +1,26 @@
+  type: config_object
+  mapping:
+    name:
+      type: varchar
```
I came across this thread 
[https://stackoverflow.com/questions/41136136/what-is-the-difference-between-varchar-and-string-in-sqlalchemys-data-type](url)

Since i am having length constraints on my fields that's why i used varchar here

vivekpan27 replied:
"sqlalchemy's data type"?
But this is Drupal no?
I don't think they are related.

sandipta18 replied:
Ok i've changed them

vivekpan27 commented on web/modules/custom/configform/src/DatabaseService.php:
```diff
@@ -0,0 +1,44 @@
+/**
+ * Instantaniates database connection and inserts data.
+ */
+class DatabaseService {
```
If you have added this service just to create a wrapper function for the database::insert() then this is not required.

vivekpan27 replied:
I don't know how it was previously, but this class is not needed.

sandipta18 replied:
OK, i'll remove it

sandipta18 replied:
> I don't know how it was previously, but this class is not needed.

Previously it was :
`\Drupal::database()->insert('configform_example')->fields([
      'email' => $values['email'],
      'name' =>  $values['name'],
      'password' => md5($values['password'])
    ])->execute();`

sandipta18 replied:
Fixed

vivekpan27 commented on web/modules/custom/configform/src/Form/ConfigForm.php:
```diff
@@ -0,0 +1,251 @@
+  public function submitDataAjax(array &$form, FormFormStateInterface $form_state) {
+    $ajax_response = new AjaxResponse();
+    $output = $this->validate($form_state);
+    if ($output === TRUE) {
```
```
if ($output) {
```

vivekpan27 commented on web/modules/custom/configform/src/Form/ConfigForm.php:
```diff
@@ -0,0 +1,251 @@
+      $ajax_response->addCommand(new CssCommand('#error-message', ['display' => 'none']));
+    }
+    else {
+      // If validatiion failed showing the error message.
```
typo on validation

vivekpan27 commented on web/modules/custom/configform/src/Form/ConfigForm.php:
```diff
@@ -0,0 +1,251 @@
+    }
+    $subscribe_value = $form_state->getValue('subscribe');
+    if ($subscribe_value === 'no' && empty($form_state->getValue('subscribe_message'))) {
+      return $this->t('Please give a reason for not subscribing');
```
I don't think any website asks a reason for not subscribing :)

vivekpan27 commented on web/modules/custom/configform/src/Form/SimpleForm.php:
```diff
@@ -0,0 +1,155 @@
+ * Implements simpleform.
+ */
+class SimpleForm extends FormBase {
+
+
```
Unnecessary empty lines.

vivekpan27 commented on web/modules/custom/configform/src/Form/SimpleForm.php:
```diff
@@ -0,0 +1,155 @@
+   * @param \Drupal\configform\DatabaseService $database
+   *   The database service handler.
+   */
+  public function __construct(MessengerInterface $messenger, RequestStack $request_stack, DatabaseService $database) {
```
Please split the arguments into separate lines.

vivekpan27 commented on web/modules/custom/configform/src/Form/SimpleForm.php:
```diff
@@ -0,0 +1,155 @@
+   * This function facilitates submissions of form without browser refresh .
+   *
+   * @param array $form
+   *   It conains all the fields in an associative array format.
```
Typo on contains

vivekpan27 commented on web/modules/custom/configform/src/Form/SimpleForm.php:
```diff
@@ -0,0 +1,155 @@
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   It holds the current state of the form along with the data.
+   *
+   * @return Response
```
namespace?

vivekpan27 commented on web/modules/custom/configform/config/schema/configform.schema.yml:
```diff
@@ -0,0 +1,23 @@
+      label: 'Subscription'
+    subscription_message:
+      type: string
+      label: 'Subscription Message'
```
What about allowed_domains?

vivekpan27 commented on web/modules/custom/config_form/src/Form/ConfigForm.php:
```diff
@@ -0,0 +1,251 @@
+use Drupal\Core\Ajax\HtmlCommand;
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Form\FormStateInterface as FormFormStateInterface;
```
Why "as"?

vivekpan27 commented on web/modules/custom/config_form/src/Form/ConfigForm.php:
```diff
@@ -0,0 +1,251 @@
+    // Pop function removes the last element from an array.
+    $domain = array_pop($parts);
+    // Validating based on conditions.
+    if (!filter_var($email, FILTER_VALIDATE_EMAIL)) {
```
Can we use switch case?

sandipta18 replied:
We can , but how would that make the code efficient? Because the time complexity will still be the same that is O(n)
